sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/implementation/RepositoryHandler.java

```diff
@@ -83,22 +83,19 @@ public RepositoryHandler(URI repositoryUri, ModelsRepositoryAPIImpl protocolLaye
 
             if (repositoryMetadata != null) {
                 tryFromExpanded = repositoryMetadata
-                    .flatMap(repo -> {
-                        if (repo != null && repo.getDefinition() != null
-                            && repo.getDefinition().getFeatures() != null
-                            && repo.getDefinition().getFeatures().isExpanded()) {
-                            return Mono.just(true);
-                        }
-                        return Mono.just(false);
-                    })
-                    .defaultIfEmpty(false)
-                    .block();
+                    .map(repo -> (
+                        repo != null && repo.getDefinition() != null
+                        && repo.getDefinition().getFeatures() != null
+                        && repo.getDefinition().getFeatures().isExpanded()
+                        )
+                    )
+                    .defaultIfEmpty(false);
             }
         }
 
         logger.info(String.format(StatusStrings.PROCESSING_DTMIS, targetDtmi));
 
-        return modelFetcher.fetchModelAsync(targetDtmi, repositoryUri, tryFromExpanded, context)
+        return tryFromExpanded.flatMap( t -> modelFetcher.fetchModelAsync(targetDtmi, repositoryUri, t, context))
```

Comment:
extra space at the beginning - also give it a more meaningful name instead of `t`

Reply:
Addressed